lib/dor/models/identifiable.rb

```diff
@@ -233,7 +233,20 @@ def adapt_to_cmodel
       if object_class
         self.instance_of?(object_class) ? self : self.adapt_to(object_class)
       else
```

Comment:
The `foxml_empty` test I added has no identityMetadata and falls through to the default as expected.